dsa-solutions/lc-solutions/0000-0099/0048-Rotate-Image.md

```diff
@@ -0,0 +1,215 @@
+
+        starting_row = 0
+        starting_col = 0
+        ending_row = row - 1
+        ending_col = col - 1
+
+        while count < total:
+            #starting row -> ending col
+            for index in range(starting_col, ending_col + 1):
+                if count < total:
+                    ans.append(matrix[index][ending_row])
+                    count += 1
+            ending_row -= 1
+
+            # ending col -> ending row
+            for index in range(starting_row, ending_row + 1):
+                if count < total:
+                    ans.append(matrix[ending_col][index])
+                    count += 1
+            ending_col -= 1
+
+            # ending row -> starting col
+            for index in range(ending_col, starting_col - 1, -1):
+                if count < total:
+                    ans.append(matrix[index][starting_row])
+                    count += 1
+            starting_row += 1
+
+            # starting column -> starting row
+            for index in range(ending_row, starting_row - 1, -1):
+                if count < total:
+                    ans.append(matrix[starting_col][index])
+                    count += 1
+            starting_col += 1
+
+        return ans
+```
+#### Java
+```
+import java.util.ArrayList;
+import java.util.List;
+
+class Solution {
+    public List<Integer> rotate(int[][] matrix) {
+        List<Integer> ans = new ArrayList<>();
+
+        // number of rows
+        int row = matrix.length;    
+        // number of columns
+        int col = matrix[0].length; 
+
+        int count = 0;         
+        int total = row * col; 
+
+        int startingRow = 0;
+        int startingCol = 0;
+        int endingRow = row - 1;
+        int endingCol = col - 1;
+
+        while (count < total) {
+            // starting row -> ending col
+            for (int index = startingCol; count < total && index <= endingCol; index++) {
+                ans.add(matrix[index][endingRow]);
+                count++;
+            }
+            endingRow--;
+
+            // ending col -> ending row
+            for (int index = startingRow; count < total && index <= endingRow; index++) {
+                ans.add(matrix[endingCol][index]);
+                count++;
+            }
+            endingCol--;
+
+            // ending row -> starting col
+            for (int index = endingCol; count < total && index >= startingCol; index--) {
+                ans.add(matrix[index][startingRow]);
+                count++;
+            }
+            startingRow++;
+
+            // starting column -> starting row
+            for (int index = endingRow; count < total && index >= startingRow; index--) {
+                ans.add(matrix[startingCol][index]);
+                count++;
+            }
+            startingCol++;
+        }
+
+        return ans;
+    }
+}
+```
+### Conclusion
+The above solution will help you to rotate an 2D Array (Image) with
+- Time Complexity: O(n^2)
```

Comment:
replace `O(n^2)` to `$O(n^2)$` and `O(1)` to `$O(1)$`

Reply:
Done ✅